src/librustc_passes/loops.rs

```diff
@@ -57,6 +58,11 @@ impl<'a, 'hir> Visitor<'hir> for CheckLoopVisitor<'a, 'hir> {
             hir::ExprKind::Loop(ref b, _, source) => {
                 self.with_context(Loop(source), |v| v.visit_block(&b));
             }
+            hir::ExprKind::Closure(_, ref function_decl, b, _,
+                                   Some(GeneratorMovability::Static)) => {
```

Comment:
I'd prefer using a `match` here on `movability` so that the non-diagnostic logic isn't repeated.

Reply:
Good spot